### PR TITLE
Sync v2-3-stable with v2-3-test to release 2.3.4

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.3.4rc1`!

### CODE REVIEW

**Review by Kinceo1 — APPROVED** [review]
💡

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
I looked at all "airflow" changes and nothing unexpected popped up.

**Comment by potiuk** [comment]
I've also run diff `git diff v2-3-test a4e38978194ef46565bc1e5ba53ecc65308d09aa -- dev .github/ scripts` to compare the dev/CI scripts in v2-3-test to the main before last dev change merged today with parallel runs output and did not see anything unexpected either.

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -22,7 +22,99 @@
 .. towncrier release notes start
 
 
-Airflow 2.3.3 (2022-07-05)
+Airflow 2.3.4 (2022-08-19)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
Airflow 2.3.4 (2022-08-22)
```

**Review by ashb — APPROVED** [review]

**Pull request merged**